### PR TITLE
feat(integrations): Add ticket actions to dowpdown

### PR DESCRIPTION
**Pull request by mgaeta**
This PR adds "create ticket" action options with Jira and Azure DevOps for Issue Alert Rules.
The new `EventAction`s are just skeletons and will be implemented in the following PRs. They're just here with the corresponding forms to populate new dropdown options.

Screen Shot 2020-10-23 at 3 28 43 PM

### CODE REVIEW

File: src/sentry/integrations/vsts/notify_action.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        super(AzureDevopsCreateTicketAction, self).__init__(*args, **kwargs)
+        # TODO 2.1 Add form_fields
+        self.form_fields = {}
+
+    def is_enabled(self):
+        # TODO move to superclass
+        return self.get_integrations().exists()
+
+    def after(self, event, state):
+        integration_id = None
```
Comment (scefali) [inline src/sentry/integrations/vsts/notify_action.py:40]: @mgaeta should this be part of the PR if this isn't actually going to be implemented?

File: src/sentry/integrations/jira/notify_action.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    label = u"TODO Create a {name} Jira ticket"
+    prompt = "Create a Jira ticket"
+
+    def __init__(self, *args, **kwargs):
+        super(JiraCreateTicketAction, self).__init__(*args, **kwargs)
+        # TODO 1.1 Add form_fields
+        self.form_fields = {}
+
+    def is_enabled(self):
+        # TODO move to superclass
```
Comment (scefali) [inline src/sentry/integrations/jira/notify_action.py:36]: @mgaeta do you mean the `EventAction` class?

Comment (mgaeta) [reply src/sentry/integrations/jira/notify_action.py]: I made a intermediate class `IntegrationEventAction` that DRYs up some of the logic. If it's overkill, I'll revert.

File: src/sentry/integrations/vsts/notify_action.py (hunk 1 above)
Comment (mgaeta) [reply src/sentry/integrations/vsts/notify_action.py:37]: I'll pull it out.

File: src/sentry/integrations/pagerduty/notify_action.py [hunk 3]
```diff
@@ -62,10 +62,11 @@ def clean(self):
         return cleaned_data
 
 
-class PagerDutyNotifyServiceAction(EventAction):
+class PagerDutyNotifyServiceAction(IntegrationEventAction):
```
Comment (scefali) [inline src/sentry/integrations/pagerduty/notify_action.py:65]: @mgaeta did you test these changes with all the integrations that were modified?

Comment (mgaeta) [reply src/sentry/integrations/pagerduty/notify_action.py:64]: I wasn't able to install PagerDuty, Slack, or MS Teams locally to test. It looks like existing tests pass though.

Comment (scefali) [reply src/sentry/integrations/pagerduty/notify_action.py:64]: @mgaeta I can try to help you get those running if you'd like. If not, then we should just not make the changes for those integrations and put a TODO to tackle later. Don't want to risk a change for critical functionality that we aren't testing. I'm not 100% sure that existing tests would catch any potential issue here.

Comment (mgaeta) [reply src/sentry/integrations/pagerduty/notify_action.py]: I've pulled out the refactor commits into this PR: https://github.com/getsentry/sentry/pull/21646.

**Review by scefali — APPROVED** [review]

**Pull request merged**